server/app/services/seeding/DatabaseSeedTask.java

```diff
@@ -90,16 +95,25 @@ public final class DatabaseSeedTask {
 
   private final QuestionService questionService;
   private final VersionRepository versionRepository;
+  private final CategoryRepository categoryRepository;
   private final Database database;
+  private final Environment environment;
 
   @Inject
-  public DatabaseSeedTask(QuestionService questionService, VersionRepository versionRepository) {
+  public DatabaseSeedTask(
+      QuestionService questionService,
+      VersionRepository versionRepository,
+      CategoryRepository categoryRepository,
+      Environment environment) {
     this.questionService = checkNotNull(questionService);
     this.versionRepository = checkNotNull(versionRepository);
+    this.categoryRepository = checkNotNull(categoryRepository);
     this.database = DB.getDefault();
+    this.environment = checkNotNull(environment);
   }
 
   public ImmutableList<QuestionDefinition> run() {
+    if (categoryRepository.listCategories().isEmpty()) seedProgramCategories();
```

Comment:
- (blocking) Please add curly braces to the `if`
- (non-blocking, suggestion) Move the `if` into `seedProgramCategories` and have it return first if empty. It'll make the list of items in this `run` method easier to follow.
- (question) Wanted to double check that categories go in before questions.

Reply:
- _(blocking) Please add curly braces to the if_. --> Done.
- _(non-blocking, suggestion) Move the if into seedProgramCategories and have it return first if empty. It'll make the list of items in this run method easier to follow._ --> I went back and forth on this, but put the if-statement in `seedProgramCategories` since I agree that it makes the `run()` easier to read.
- _(question) Wanted to double check that categories go in before questions._ --> It shouldn't matter that categories go in before questions because there aren't any relationships between them.